tests/Faker.Tests/Common/NameTests.cs

```diff
@@ -25,5 +25,19 @@ public void Should_Get_Suffix()
 
 			Assert.That(new[] { suffix }, Is.SubsetOf(possibleSuffixes));
 		}
+
+		[Test]
+		[Repeat(100000)]
+		public void Compute_Time_Elapsed_By_The_First_Method()
+		{
+			Name.First();
```

Comment:
Test repeated using Stopwatch and calling 100.000 time the `First()` method. Results:

```
Test Name:	Compute_Time_Elapsed_By_The_First_Method
Test Outcome:	Passed
Result StandardOutput:	
Elapsed time without caching 18487msec
Elapsed time with caching 24msec
```
Good, uh? :-P
Everything is reply-able in the last commit. Now there is a `First()` (default resource caching) and a `FirstCached()` (caches the splitted array) method. Obviously, the latter has been defined only for testing purposes.